### ISSUE TITLE
Decide how to expose parameters for question and qnaId

### ISSUE THREAD

**Issue by heaths** [Azure/azure-sdk-for-net#22511]
@KrzysztofCwalina @tg-msft @schaabs it came up in review that we should move required parameters out of `KnowledgebaseQueryOptions` and put them as required parameters, which is either `string question` or `int qnaId`. Agreed? How does that differ from `EncryptParameters` for Key Vault, though?

Also, since not all languages support overloads, I propose adding just a `GetQuestionAnswer(string projectName, int qnaId, string deploymentName = null)`. when a `qnaId` is passed, it's the only option that matters.

/cc @annatisch @rokulka

_Originally posted by @heaths in https://github.com/Azure/azure-sdk-for-net/pull/21810#discussion_r664912422_

**Comment by heaths** [comment]
@annatisch or @johanste could you summarize what you're thinking for Python? For .NET, I'm leaning toward:

```csharp
Response<KnowledgebaseAnswers> QueryKnowledgebase(
  string projectName,
  string question,
  QueryKnowledgebaseOptions options = default,
  CancellationToken cancellationToken = default);

Response<KnowledgebaseAnswers> GetQuestionAnswers(
  string projectName,
  int qnaId,
  CancellationToken cancellationToken = default);
```

/cc @tg-msft @KrzysztofCwalina @rokulka

**Comment by annatisch** [comment]
I think I'd be okay with moving `qnaId` into a separate path/API.
In doing so - we could remove it from the `KnowledgebaseQueryParameters` object - allowing `question` be become a required parameter and simplifying the usability of that API.
Given that `question` is part of the same API body - I would prefer to keep in in the parameters object thought - this follows better with our new Python designs for passing around JSON.

So for Python, we could update the `query_knowledgebase` API to something like:
```python

    @overload
    def query_knowledgebase(
            self,
            parameters: KnowledgebaseQueryParameters,
            *,
            project_name: str,
            deployment_name: Optional[str] = None,
            **kwargs
    ) -> KnowledgebaseAnswers:
        ...

    @overload
    def query_knowledgebase(
            self,
            *,
            project_name: str,
            question: str,
            top: Optional[int] = None,
            user_id: Optional[str] = None,
            confidence_score_threshold: Optional[float] = None,
            context: Optional["KnowledgebaseAnswerRequestContext"] = None,
            ranker_type: Optional[Union[str, "RankerType"]] = None,
            strict_filters: Optional["StrictFilters"] = None,
            answer_span_request: Optional["AnswerSpanRequest"] = None,
            include_unstructured_sources: Optional[bool] = None,
            deployment_name: Optional[str] = None,
            **kwargs
    ) -> KnowledgebaseAnswers:
        ...

    def query_knowledgebase(self, *args, **kwargs):
        ...
```

**Comment by annatisch** [comment]
@johanste - how do you feel about the overloads above?
It should be technically possible to generate that in autorest.

**Comment by johanste** [comment]
Gut reaction, I think the overloads look good. Let's try them out!

**Comment by heaths** [comment]
Per offline discussion, "Parameters" -> "Options" and, at least for .NET, we'll use only an Options class that will take required parameters in their constructor (`question` or `qnaId`) and have overloads matching Text Analytics (method name itself is under discussion):

```csharp
public virtual Response<KnowledgebaseAnswers> QueryKnowledgebase(
  QueryKnowledgebaseOptions options,
  CancellationToken cancellationToken = default);

public virtual Response<TextAnswers> QueryText(
  string question,
  IEnumerable<string> records,
  CancellationToken cancellationToken = default);

public virtual Response<TextAnswers> QueryText(
  string question,
  IEnumerable<TextInput> records,
  CancellationToken cancellationToken = default);

public virtual Response<TextAnswers> QueryText(
  QueryTextOptions options,
  CancellationToken cancellationToken = default);
```

```csharp
public class QueryKnowledgebaseOptions
{
  public QueryKnowledgebaseOptions(string question);
  public QueryKnowledgebaseOptions(int qnaId);
  public string Question { get; }
  public int QnaId { get; }
  // Other properties are get + set, but ignored for qnaId by service
}
```

**Comment by annatisch** [comment]
@heaths - how do you feel about moving `QnaId` out of the Options class into a separate path/API?

**Comment by heaths** [comment]
I asked @rokulka previously about that - something like a `GetQuestionAnswer(int qnaId)` - and he thought it wasn't useful enough to stand alone, which is why they designed the service endpoint as they did.

That said, apart from "Parameters" -> "Options", if you were to do what you prototyped above with having a `dict`-like bag as well, having an overload that takes `qnaId` as well as `question` does seem to fit with Python's guidelines. In that case, it probably doesn't make sense to include other properties of `QueryKnowledgebaseOptions` in the `dict`-like API.

**Comment by annatisch** [comment]
Yup that's fair. I can make this work in the overloads above.

**Comment by heaths** [comment]
This was completed for .NET: https://docs.microsoft.com/dotnet/api/azure.ai.language.questionanswering.models.queryknowledgebaseoptions.-ctor?view=azure-dotnet-preview
